skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.kt

```diff
@@ -151,50 +133,10 @@ open class SkiaLayer : HardwareLayer() {
         )
     }
 
-    private fun initSkija() {
-        initRenderTarget()
-        initSurface()
-    }
-
-    private fun initRenderTarget() {
-        skijaState.apply {
-            clear()
-            val dpi = contentScale
-            val width = (width * dpi).toInt().coerceAtLeast(0)
-            val height = (height * dpi).toInt().coerceAtLeast(0)
-            renderTarget = when (api) {
-                GraphicsApi.OPENGL -> {
-                    val gl = OpenGLApi.instance
-                    val fbId = gl.glGetIntegerv(gl.GL_DRAW_FRAMEBUFFER_BINDING)
-                    makeGLRenderTarget(
-                        width,
-                        height,
-                        0,
-                        8,
-                        fbId,
-                        FramebufferFormat.GR_GL_RGBA8
-                    )
-                }
-                GraphicsApi.METAL -> makeMetalRenderTarget(
-                    width,
-                    height,
-                    0
-                )
-                else -> TODO("Unsupported yet")
-            }
-        }
-    }
-
-    private fun initSurface() {
-        skijaState.apply {
-            surface = Surface.makeFromBackendRenderTarget(
-                context!!,
-                renderTarget!!,
-                SurfaceOrigin.BOTTOM_LEFT,
-                SurfaceColorFormat.RGBA_8888,
-                ColorSpace.getSRGB()
-            )
-            canvas = surface!!.canvas
-        }
+    private fun fallbackToRaster() {
+        println("Falling back to software rendering...")
+        skijaState = RasterContextHandler(this)
+        redrawer = RasterRedrawer(this)
```

Comment:
We should dispose previous redrawer:
```
redrawer?.dispose()
```
probably we should also call 'skijaState.dispose()'

Reply:
And probably 'draw' isn't good place to dispose redrawer (but we should do it anyway).

Because on macOs we will dispose AWTGLLayer when we draw inside AWTGLLayer.draw. Windows/Linux should be fine.

Reply:
I have an idea:
```
private fun scheduleFallbackToRaster() {
    if (!isFallbackScheduled) {
        isFallbackScheduled = true
        SwingUtilities.invokeLater {
            if (!isDisposed) {
                fallbackToRaster()
            }
        }
    }
}
```

Reply:
Issue: https://github.com/JetBrains/skiko/issues/57